drivers/pwm/pwm_renesas_ra8.c

```diff
@@ -0,0 +1,590 @@
+/*
+ * Copyright (c) 2024 Renesas Electronics Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include <zephyr/kernel.h>
+#include <zephyr/device.h>
+#include <zephyr/irq.h>
+#include <zephyr/drivers/clock_control/renesas_ra_cgc.h>
+#include <zephyr/drivers/pwm.h>
+#include <zephyr/drivers/pinctrl.h>
+#include "r_gpt.h"
+#include "r_gpt_cfg.h"
+#include <zephyr/logging/log.h>
+#include <stdio.h>
+
+LOG_MODULE_REGISTER(pwm_renesas_ra8, CONFIG_PWM_LOG_LEVEL);
+
+#define DT_DRV_COMPAT renesas_ra8_pwm
+
+#define MAX_PIN                                       (2U)
+#define GPT_PRV_GTIO_HIGH_COMPARE_MATCH_LOW_CYCLE_END (0x6U)
+#define GPT_PRV_GTIO_LOW_COMPARE_MATCH_HIGH_CYCLE_END (0x9U)
+#define GPT_PRV_GTIOR_INITIAL_LEVEL_BIT               (4)
+#define GPT_PRV_GTIO_TOGGLE_COMPARE_MATCH             (0x3U)
+
+static const char *const ra8_pwm_errmsg[] = {
+	[FSP_ERR_ASSERTION] = "[FSP_ERR_ASSERTION] A required input pointer is NULL "
+			      "OR the source divider is invalid",
+	[FSP_ERR_ALREADY_OPEN] = "[FSP_ERR_ALREADY_OPEN] Module is already open",
+	[FSP_ERR_IRQ_BSP_DISABLED] =
+		"[FSP_ERR_IRQ_BSP_DISABLED] ISR must be enabled to use callback",
+	[FSP_ERR_INVALID_MODE] =
+		"[FSP_ERR_INVALID_MODE] Triangle wave PWM is only supported if "
+		"GPT_CFG_OUTPUT_SUPPORT_ENABLE is 2 "
+		"OR Selected channel does not support external count sources "
+		"OR External and event count sources not are available in this mode",
+	[FSP_ERR_NOT_OPEN] = "[FSP_ERR_NOT_OPEN] The instance is not opened",
+	[FSP_ERR_INVALID_ARGUMENT] = "[FSP_ERR_INVALID_ARGUMENT] Duty cycle is larger than period",
+	[FSP_ERR_UNSUPPORTED] = "[FSP_ERR_UNSUPPORTED] PLL GPT_CFG_OUTPUT_SUPPORT_ENABLE is 0"};
+
+#define ZEPHYR_ERROR_LOG(err)                                                                      \
+	{                                                                                          \
+		if (err == FSP_SUCCESS) {                                                          \
+			(void)0;                                                                   \
+		} else {                                                                           \
+			LOG_ERR("%s: %s", __func__, ra8_pwm_errmsg[err]);                          \
+		}                                                                                  \
+	}
```

Comment:
that's a pretty messed up way of handling errors and very uncommon thoughout the code base, why the (void)0 rather than swap the condition and what's the point of continuing on errors?

It also seems unnecessarily verbose, do you realy want to use the device flash for all these errors? Just check the error code after every call and log the code, it's an embedded driver it does not need to be user friendly.

Reply:
I removed this err define, use only the LOG_ERR in our source. Thanks for your comment